### DISCUSSION TITLE
[Bug]Why  are there unequal spatial lag results in creating a moran scatter plot and using the calculator？

### DISCUSSION THREAD

**Discussion by 0xashes** [GeoDaCenter/geoda#2428] [Q&A]
**Describe the bug**
Why are there unequal spatial lag results in creating a moran scatter plot and using the calculator？
When I compute the spatial lag, getting it by creating a moran scatter is a normal easy way. And also there is a way that I calculate the spatial lag using the calculator with a row-standarlized weights(see http://geodacenter.github.io/workbook/5a_global_auto/lab5a.html). 
I can always get the unequal results once the spatial weights were built by  a kernel function except the uniform func.
![image](https://user-images.githubusercontent.com/67677288/198838306-b56a82ff-4d15-46f1-a463-01524f39d76e.png)

The calculator results are shown in the '111' column. 

Expect answers.

**Comment by lanselin** [comment]
GeoDa does not use actual weights in its analyses. So a kernel weights matrix in a Moran scatter plot is used only to identify the
neighbors, the actual weights are ignored. This is the case for all spatial weights files that include weights information (GWT, KWT).
In contrast, the calculator allows one to compute the spatial lag using the actual weights, e.g., inverse distance weights and
kernel weights. That is why the lag “saved” from the Moran scatter plot and the lag computed in the calculator are different.
This is by design, it is not a bug.

See the documentation of the Moran scatter plot:

The current implementation of the Moran scatter plot functionality in GeoDa ignores the specific values for the weights and only takes into account the presence of connectivity (i.e., non-zero weights). This is the approach taken for all spatial autocorrelation analyses in GeoDa. As a result, in the computation of the spatial lag in the Moran scatter plot, all the neighbors receive the same weight.

**Comment by 0xashes** [comment]
> GeoDa does not use actual weights in its analyses. So a kernel weights matrix in a Moran scatter plot is used only to identify the neighbors, the actual weights are ignored. This is the case for all spatial weights files that include weights information (GWT, KWT). In contrast, the calculator allows one to compute the spatial lag using the actual weights, e.g., inverse distance weights and kernel weights. That is why the lag “saved” from the Moran scatter plot and the lag computed in the calculator are different. This is by design, it is not a bug. See the documentation of the Moran scatter plot: The current implementation of the Moran scatter plot functionality in GeoDa ignores the specific values for the weights and only takes into account the presence of connectivity (i.e., non-zero weights). This is the approach taken for all spatial autocorrelation analyses in GeoDa. As a result, in the computation of the spatial lag in the Moran scatter plot, all the neighbors receive the same weight.
> […](#)
> On Oct 29, 2022, at 11:06 AM, skyknoow ***@***.***> wrote: Describe the bug Why are there unequal spatial lag results in creating a moran scatter plot and using the calculator？ When I compute the spatial lag, getting it by creating a moran scatter is a normal easy way. And also there is a way that I calculate the spatial lag using the calculator with a row-standarlized weights(see http://geodacenter.github.io/workbook/5a_global_auto/lab5a.html <http://geodacenter.github.io/workbook/5a_global_auto/lab5a.html>). I can always get the unequal results once the spatial weights were built by a kernel function except the uniform func. <https://user-images.githubusercontent.com/67677288/198838306-b56a82ff-4d15-46f1-a463-01524f39d76e.png> The calculator results are shown in the '111' column. Expect answers. — Reply to this email directly, view it on GitHub <#2405>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA6STH6S7VAJOM7A2Z3ELNTWFU4PFANCNFSM6AAAAAARR2OR3Q>. You are receiving this because you are subscribed to this thread.

Thanks for your detailed answer. This completely solved my doubts.
